mlrun/api/api/utils.py

```diff
@@ -150,12 +156,169 @@ async def submit_run(db_session: Session, auth_info: mlrun.api.schemas.AuthInfo,
+                    logger.warning(
+                        "Found volume without name, skipping obfuscation", volume=volume
+                    )
+                    continue
+                volume_name = get_item_attribute(volume, "name")
+                # Best effort - usually we'll have a volume mount to /users/<username>, try to resolve from there
+                # if not found, skip obfuscation
+                username = None
+                found_more_than_one_username = False
+                for volume_mount in volume_name_to_volume_mounts[volume_name]:
+                    sub_path = get_item_attribute(
+                        volume_mount, "subPath"
+                    ) or get_item_attribute(volume_mount, "sub_path")
+                    if sub_path and sub_path.startswith("users/"):
+                        username_from_sub_path = sub_path.replace("users/", "")
+                        if username_from_sub_path:
+                            if (
+                                username is not None
+                                and username != username_from_sub_path
+                            ):
+                                logger.warning(
+                                    "Found more than one user for volume, skipping obfuscation",
+                                    volume=volume,
+                                    volume_mounts=volume_name_to_volume_mounts[
+                                        volume_name
+                                    ],
+                                )
+                                found_more_than_one_username = True
+                                break
+                            username = username_from_sub_path
+                if found_more_than_one_username:
+                    continue
+                if not username:
+                    logger.warning(
+                        "Could not resolve username from volume mount, skipping obfuscation",
+                        volume=volume,
+                        volume_mounts=volume_name_to_volume_mounts[volume_name],
+                    )
+                    continue
+                username: str
+                secret_name = mlrun.api.crud.Secrets().store_auth_secret(
+                    mlrun.api.schemas.AuthSecretData(
+                        provider=mlrun.api.schemas.SecretProviderName.kubernetes,
+                        username=username,
+                        access_key=access_key,
+                    )
+                )
+                access_key_attribute_name = "accessKey"
+                secret_ref_attribute_name = "secretRef"
+                flex_volume_attribute_name = "flexVolume"
+                if get_item_attribute(volume, "flex_volume"):
+                    flex_volume_attribute_name = "flex_volume"
+                    access_key_attribute_name = "access_key"
+                    secret_ref_attribute_name = "secret_ref"
```

Comment:
I think this knowledge of mapping the keys and attribute name of volume type would be helpful not only for this function, I would move it to some class that maps those keys for each volume type like you did in `AuthSecretData`, what do you think?

Reply:
I changed the code to be much more simpler by simply sanitizing all volumes at the beginning, take a look